### PR TITLE
Doc: Custom markdown style guide

### LINKED ISSUE #352: [📑 Docs]: It will be great to create a Markdown style guide with the custom annotations we have
### What Docs changes are you proposing?

I'd like to propose creating a md style guide documenting all the custom tags we have working in our website so the people can create better docs and we avoid creating unnecessary new tags. This improve discoverability and contributors experience.

- Example : https://github.com/couchbaselabs/docs-style-guide/blob/master/md-style-guide.md
- All the custom tags could be identified in this file: components/StyledMarkdown.tsx 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by DhairyaMajmudar**
**GitHub Issue:** #352
closes #352  

**Summary**:
Added custom markdown writing guide in contributing.md file.

**Do you think resolving this issue might require an [Architectural Decision Record (ADR)](https://github.com/json-schema-org/community/blob/main/CONTRIBUTING.md#key-architectural-decisions)? (significant or noteworthy)**

No

### CODE REVIEW

**Comment by benjagm** [comment]
> @benjagm pls. review the PR

Thanks a lot for this PR! Ideally I'd like this guide to be a separate file called md-style-guide.md adding a link to it in the contributing.md.

File: pages/understanding-json-schema/structuring.md [hunk 1]
```diff
@@ -1,8 +1,7 @@
 ---
 title: "Structuring a complex schema"
 section: docs
----
-
+---
```
Comment (benjagm) [inline pages/understanding-json-schema/structuring.md:4]: Why editing this file?

**Review by benjagm — CHANGES_REQUESTED** [review]
Thanks a lot for this PR! 

Ideally I'd like this guide to be a separate file called md-style-guide.md adding a link to it in the contributing.md.

File: pages/understanding-json-schema/structuring.md (hunk 1 above)
Comment (DhairyaMajmudar) [reply pages/understanding-json-schema/structuring.md:4]: I think its done by my prettier extension , I will resolve this 😅

**Comment by DhairyaMajmudar** [comment]
@benjagm I have done the suggested changes pls. have a look

File: md-style-guide.md
[the remark this answers is not in the document]
Comment (DhairyaMajmudar) [reply md-style-guide.md:1]: sounds good to me , I will modify it more

**Review by benjagm — CHANGES_REQUESTED** [review]
I agree with @gregsdennis . We need to develop the content of style guide to provide context of the document and clear guidance and examples.

**Comment by DhairyaMajmudar** [comment]
@benjagm I have updated my PR with suggested changes, can you pls. provide the screenshot for `blockquote` from the website.

File: md-style-guide.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+### **Star**
+
+#### Adds a star with a label along with it. In place of `"label"` text can be added which needs to be shown. 
+
+```markdown
+<Star label="label"/>
+```
+
+#### Image
+![image](https://github.com/DhairyaMajmudar/json-schema-website/assets/124715224/936f4561-a37e-42c3-aaad-8ac463ae5d34)
```
Comment (gregsdennis) [inline md-style-guide.md:12]: I'd prefer not to have images but to actual show the markdown working.  So you'll have a `<Star/>` tag in a code block (as you do), then you'd actually have another tag not in a code block so it renders.

The idea is to show it working and show how it works.

**Review by gregsdennis — CHANGES_REQUESTED** [review]

File: md-style-guide.md (hunk 2 above)
Comment (benjagm) [reply md-style-guide.md:12]: It will only render in the website, GitHub or other markdown renders won't be able to interpret our Custom Tags

Comment (DhairyaMajmudar) [reply md-style-guide.md:12]: Gotcha thanks for letting me know :D

Comment (benjagm) [reply md-style-guide.md:12]: I had a quick chat with @gregsdennis and we agreed that the best place for this guide is inside the pages folder to be considered as a website page instead as a regular md file in the root folder, this way we can have everything rendered nicely. We envision this page to be accesed as part of the website experience. 

We can add the link to this page in the text part of the header of the blog:
Screenshot 2024-02-28 at 21 41 17

Comment (benjagm) [reply md-style-guide.md:12]: @DhairyaMajmudar Thanks for the flexibility addressing additional requirements!

**Comment by DhairyaMajmudar** [comment]
@benjagm I have added the file in pages folder and also did the changes as suggested by @gregsdennis , now what should to done to render the .md file in the website ?

**Review by gregsdennis — APPROVED** [review]
Yes!  This is what I'm looking for.

Can I get a screenshot of a section of the page, please?  Then whatever needs to be done to ensure the page is published.  But I'm happy with this.

**Comment by DhairyaMajmudar** [comment]
Sure attaching some screenshots 

Screenshot from 2024-03-01 01-11-53

Screenshot from 2024-03-01 01-12-07

**Comment by gregsdennis** [comment]
Looking at the site, I'm inclined to agree with @benjagm that it needs to be added to the "want to publish" text on the blog.  I can't really see anywhere else for it to go.

**Comment by gregsdennis** [comment]
@benjagm this needs your refreshed review.

I'm still split between including this in the blog header (where you suggested) or somewhere in docs.  I'm unsure because it applies to the whole site.  Regardless, it needs to be easily discoverable.

**Comment by gregsdennis** [comment]
What's the URL path where this appears?

**Comment by DhairyaMajmudar** [comment]
> What's the URL path where this appears?

The page is accessible at `http://localhost:3000/md-style-guide`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -21,6 +21,7 @@ If you don't have time to contribute, that's fine. There are other easy ways to
   - 📋 [Improving the Documentation](#-improving-the-documentation)
   - 🌐 [Improving the website](#-improving-the-website)
   - 🎨 [Improving the Design](#-improving-the-design)
+  - 🚀 [Custom Markdown Writing Style Guide](./md-style-guide.md)
```
Comment (benjagm) [inline CONTRIBUTING.md:24]: This link to open with proper render should be https://json-schema.org/md-style-guide

In addition I think this link doesn't go as part of the list of types of contribution, instead I see this as a section H2 section before Triage with a title "Markdown Style Guide" and a short paragraph to briefly introduce this guide for users contributing to docs or other types of written content like blogs or case studies.

**Review by benjagm — CHANGES_REQUESTED** [review]
Left a small suggestion in the contributing.md.

Great work. We are very close.

File: CONTRIBUTING.md (hunk 3 above)
Comment (gregsdennis) [reply CONTRIBUTING.md:24]: We also need to consider how this page will be reachable from the website.

Comment (DhairyaMajmudar) [reply CONTRIBUTING.md:24]: > This link to open with proper render should be https://json-schema.org/md-style-guide
> 
> In addition I think this link doesn't go as part of the list of types of contribution, instead I see this as a section H2 section before Triage with a title "Markdown Style Guide" and a short paragraph to briefly introduce this guide for users contributing to docs or other types of written content like blogs or case studies.

sure would be happy to do these changes.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -78,6 +78,10 @@ Generally, pull requests should be made to the `main` branch.
 
 Most PRs, will be left open for a minimum of 14 days.  Minor fixes may be merged more quickly once approved by a project member.
 
+### Markdown Style Guide
+
+User contributing to doc and or other types of written content like blogs or case studies can have a look at our [custom markdown style guide](https://json-schema.org/md-style-guide). This style guide has information of custom markdown tags used by us.
```
Comment (benjagm) [inline CONTRIBUTING.md:83]:
```suggestion
Contributors to our Docs or our Blog can have a look at the [custom markdown style guide](https://json-schema.org/md-style-guide) with a list of useful markdown tags providing tools to easily create cool content that provides a better user experience.
```

**Review by benjagm — CHANGES_REQUESTED** [review]
Lat's details for this great contribution. I just let a suggestion.

File: CONTRIBUTING.md (hunk 4 above)
Comment (DhairyaMajmudar) [reply CONTRIBUTING.md:83]: Thank you for improving : D

**Review by benjagm — APPROVED** [review]
Let's merge it. Good job.

**Pull request merged**